pkg/placement/placement.go

```diff
@@ -53,33 +52,53 @@ const (
 //  3. for updates on label selectors, re-evaluate if existing objects should be removed
 //     from clusters.
 func (c *Controller) handlePlacement(obj runtime.Object) error {
-	placement := obj.DeepCopyObject()
+	placement, err := runtimeObjectToPlacement(obj.DeepCopyObject())
+	if err != nil {
+		return err
+	}
 
 	// handle requeing for changes in placement, excluding deletion
 	if !isBeingDeleted(obj) {
-		if err := c.requeueForPlacementChanges(); err != nil {
+		if err := c.requeueForPlacementChanges(placement); err != nil {
 			return err
 		}
+	} else {
+		c.placementResolver.DeleteResolution(placement.GetName())
 	}
 
 	if err := c.handlePlacementFinalizer(placement); err != nil {
 		return err
 	}
 
+	// TODO (maroon): this should be deleted when transport is ready
 	if err := c.cleanUpObjectsNoLongerMatching(placement); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (c *Controller) requeueForPlacementChanges() error {
+func (c *Controller) requeueForPlacementChanges(placement *v1alpha1.Placement) error {
```

Comment:
This func is now doing more than "requeue"; I suggest updating the name.

Reply:
reverted this function to how it was and moved the new logic to handlePlacement